doc/source/whatsnew/v0.23.5.txt

```diff
@@ -39,6 +39,7 @@ Bug Fixes
 **Groupby/Resample/Rolling**
 
 - Bug in :meth:`DataFrame.resample` when resampling ``NaT`` in ``TimeDeltaIndex`` (:issue:`13223`).
+- Bug in :meth:`DataFrameGroupBy.rank` and :meth:`SeriesGroupBy.rank` with empty groups (:issue:`22519`)
```

Comment:
The scope of this is only limited to when `pct=True` right? If so I think that should be indicated here. Also it's not as much the groups as the values?

Reply:
Re: scope, yep, makes sense.  Re: groups vs. values, they go hand in hand semantically.  If you have only `NaN` in your group, it's empty.